h2/utilities.py

```diff
@@ -221,8 +221,8 @@ def _reject_te(headers, hdr_validation_flags):
     its value is anything other than "trailers".
     """
     for header in headers:
-        if header[0] == b'te':
-            if header[1].lower().strip() != b'trailers':
+        if header[0] in (b'te', u'te'):
+            if header[1].lower().strip() not in (b'trailers', u'trailers'):
```

Comment:
Isn’t that the change being added in #281?

We strip extra whitespace as a normalisation step sending, and normalisation precedes validation, so we don’t need the `strip()` call. <sup>1</sup>

And we’ll reject header fields that have bad whitespace, so even if the missing `strip()` allows something to slip through here, we’ll catch it earlier/later.

<sup>
<sup>1</sup> Somebody who disables normalisation but not validation could get bitten here and allowed to send invalid headers, but that seems like a weird set of config to use, and we trust them to behave sensibly.
</sup>